azurerm/internal/services/web/resource_arm_function_app_slot.go

```diff
@@ -0,0 +1,657 @@
+										Type:         schema.TypeString,
+										Optional:     true,
+										ValidateFunc: validation.StringIsNotEmpty,
+									},
+								},
+							},
+						},
+						"min_tls_version": {
+							Type:     schema.TypeString,
+							Optional: true,
+							Computed: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								string(web.OneFullStopZero),
+								string(web.OneFullStopOne),
+								string(web.OneFullStopTwo),
+							}, false),
+						},
+						"ftps_state": {
+							Type:     schema.TypeString,
+							Optional: true,
+							Computed: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								string(web.AllAllowed),
+								string(web.Disabled),
+								string(web.FtpsOnly),
+							}, false),
+						},
+						"cors": azure.SchemaWebCorsSettings(),
+					},
+				},
+			},
+
+			"auth_settings": azure.SchemaAppServiceAuthSettings(),
+
+			"site_credential": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"username": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"password": {
+							Type:      schema.TypeString,
+							Computed:  true,
+							Sensitive: true,
+						},
+					},
+				},
+			},
+
+			"tags": tags.Schema(),
+		},
+	}
+}
+
+func resourceArmFunctionAppSlotCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Web.AppServicesClient
+	ctx, cancel := timeouts.ForCreate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	slot := d.Get("name").(string)
+	resourceGroup := d.Get("resource_group_name").(string)
+	functionAppName := d.Get("function_app_name").(string)
+
+	if features.ShouldResourcesBeImported() && d.IsNewResource() {
+		existing, err := client.GetSlot(ctx, resourceGroup, functionAppName, slot)
+		if err != nil {
+			if !utils.ResponseWasNotFound(existing.Response) {
+				return fmt.Errorf("Error checking for presence of existing Slot %q (Function App %q / Resource Group %q): %s", slot, functionAppName, resourceGroup, err)
+			}
+		}
+
+		if existing.ID != nil && *existing.ID != "" {
+			return tf.ImportAsExistsError("azurerm_function_app_slot", *existing.ID)
+		}
+	}
+
+	location := azure.NormalizeLocation(d.Get("location").(string))
+	kind := "functionapp"
+	if osTypeRaw, ok := d.GetOk("os_type"); ok {
+		osType := osTypeRaw.(string)
+		if osType == "linux" {
+			kind = "functionapp,linux"
+		}
+	}
+
+	appServicePlanId := d.Get("app_service_plan_id").(string)
+	enabled := d.Get("enabled").(bool)
+	clientAffinityEnabled := d.Get("client_affinity_enabled").(bool)
+	httpsOnly := d.Get("https_only").(bool)
+	dailyMemoryTimeQuota := d.Get("daily_memory_time_quota").(int)
+	t := d.Get("tags").(map[string]interface{})
+	appServiceTier, err := getFunctionAppServiceTier(ctx, appServicePlanId, meta)
+	if err != nil {
+		return err
+	}
+
+	basicAppSettings := getBasicFunctionAppAppSettings(d, appServiceTier)
```

Comment:
`getBasicFunctionAppAppSettings` has been updated in master and now needs an additional arg `endpointsuffix`.  If you rebase against master, you can see this implemented in `azurerm_function_app`.